src/nationalarchives/components/nested-navigation/nested-navigation.ts

```diff
@@ -0,0 +1,385 @@
+          if (countChecked === all.length) {
+            parentLI.setAttribute('aria-checked', 'true')
+            parentLI.setAttribute('aria-selected', 'true')
+          }
+          // None checked
+          if (countChecked === 0) {
+            parentLI.setAttribute('aria-selected', 'false')
+            parentLI.setAttribute('aria-checked', 'false')
+          }
+          const nextEl: HTMLUListElement | null | undefined =
+            ul.parentElement?.closest('[role=group]')
+          if (nextEl != null) {
+            this.setParentState(nextEl)
+          }
+        }
+      }
+    }
+  }
+
+  setFocusToItem: (element?: HTMLLIElement | null) => void = (element) => {
+    Array.from(this.treeItems).forEach((item) => {
+      (item as HTMLElement).tabIndex = -1
+    })
+    if (element != null) {
+      element.tabIndex = 0
+      element.focus()
+      this.currentFocus = element
+    }
+  }
+
+  setFocusToPreviousItem: (input: HTMLLIElement | null) => void = (input) => {
+    if (input != null) {
+      const li: HTMLLIElement | null = input.closest('li')
+      // Do you have a sibling
+      if (li?.previousElementSibling != null) {
+        // Does sibling have an aria-expanded=true
+        if (li.previousElementSibling.getAttribute('aria-expanded') === 'true') {
+          // Go to sibling's last child
+          const lastChild: HTMLLIElement | null = li.previousElementSibling.querySelector(':scope > ul > li:last-child')
+          if (lastChild !== null) {
+            this.setFocusToItem(lastChild)
+          }
+        } else {
+          // Go to previous sibling
+          this.setFocusToItem(li.previousElementSibling as HTMLLIElement)
+        }
+      } else if (li?.parentElement != null) {
+        // Go to parent
+        this.setFocusToItem(li.parentElement.closest('li') as HTMLLIElement)
+      }
+    }
+  }
+
+  setFocusToNextItem: (input: HTMLLIElement | null) => void = (input) => {
+    if (input != null) {
+      const li: HTMLLIElement | null = input.closest('li')
+      if (li !== null) {
+        // Do you have a child
+        if (li.getAttribute('aria-expanded') === 'true') {
+          // go to first child
+          const firstChild: HTMLLIElement | null = li.querySelector('ul > li')
+          if (firstChild !== null) {
+            this.setFocusToItem(firstChild)
+          }
+        } else {
+          if (li.nextElementSibling !== null) {
+            // Go to next sibling
+            this.setFocusToItem(li.nextElementSibling as HTMLLIElement)
+          } else if (li.parentElement !== null) {
+            // Go to parents next sibling
+            const parent: HTMLLIElement | null = li.parentElement.closest('li')
+            if (parent?.nextElementSibling != null) {
+              this.setFocusToItem(parent.nextElementSibling as HTMLLIElement)
+            }
+          }
+        }
+      }
+    }
+  }
+
+  handleKeyDown: (ev: KeyboardEvent) => void = (ev) => {
```

Comment:
This is a pretty long function, with some complex logic for some of the cases. Is it possible to move the case logic to separate functions and call those? Not sure that would make things better or worse?

Reply:
I think your right on the ArrowRight, ArrowLeft and End ones so I've moved those into separate private functions. The others are two lines so I think that's ok.